gui/gui/src/ts/components/search/SearchTextInput.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from "react";
+
+import {Paper, InputAdornment, InputBase} from "@material-ui/core";
+
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faSearch} from "@fortawesome/free-solid-svg-icons";
+
+export const SearchTextInput: React.FunctionComponent<{
+  onSubmit: (value: string) => void;
+}> = ({onSubmit}) => {
+  return (
+    <Paper variant="outlined" square>
+      <InputBase
+        inputProps={{
+          "data-cy": "searchTextInput",
+        }}
+        fullWidth
+        startAdornment={
+          <InputAdornment position="end" style={{marginRight: "8px"}}>
+            <FontAwesomeIcon icon={faSearch} />
+          </InputAdornment>
+        }
+        onKeyDown={(event: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
Nit: in terms of device portability I believe it's better to do a <form> with no action and capture its onSubmit (but remember to do e.preventDefault()). There are many ways to submit a form besides pressing enter. Some devices have keys, for example.